lib/tapioca/dsl/helpers/active_record_column_type_helper.rb

```diff
@@ -180,9 +180,16 @@ def type_for_activerecord_value(column_type)
                  defined?(ActiveRecord::ConnectionAdapters::PostgreSQL::OID::Array) &&
                    ActiveRecord::ConnectionAdapters::PostgreSQL::OID::Array === type
                }
-            "T::Array[#{type_for_activerecord_value(column_type.subtype)}]"
+            "T::Array[#{type_for_activerecord_value(column_type.subtype, null:)}]"
           else
-            ActiveModelTypeHelper.type_for(column_type)
+            base_type = ActiveModelTypeHelper.type_for(column_type)
+
+            # It's possible that when ActiveModel::Type::Value is used, the signature being reflected on in
+            # ActiveModelTypeHelper.type_for(type_value) may say the type can be nilable. However, if the type is
+            # persisted and the column is not nullable, we can assume it's not nilable.
+            return as_non_nilable_type(base_type) if @column_type_option.persisted? && !null
```

Comment:
Since you've also contributed the `EncryptedAttributeType` case in line 136, this could affect that too right? So we should return non nilable types if possible in that case too.

Reply:
Ah yes, thanks for pointing that out 👍